scripts/BadgeManager.js

```diff
@@ -2,20 +2,38 @@
 
 'use strict';
 
-function BadgeManager(addEventListener) {
+function BadgeManager(addEventListener, isEnabled) {
 
 	addEventListener("hourMusic", function(hour) {
-		chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		}
+		playIcon();
 	});
 
 	addEventListener("kkStart", function() {
-		chrome.browserAction.setBadgeText({ text: "KK" });
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: "KK" });
+		}
+		playIcon();
```

Comment:
It could be cute to change the icon to KK instead of the leaf when this event comes in.